rotkehlchen/api/v1/schemas.py

```diff
@@ -3861,3 +3861,32 @@ def make_calendar_entry(self, data: dict[str, Any], **_kwargs: dict[str, Any]) -
 
 class HistoricalPerAssetBalanceSchema(SnapshotTimestampQuerySchema, AsyncQueryArgumentSchema):
     asset = AssetField(expected_type=Asset, load_default=None)
+
+
+class HistoricalPricesPerAssetSchema(AsyncQueryArgumentSchema, TimestampRangeSchema):
+    interval = fields.Integer(strict=True, required=True)
+    asset = AssetField(expected_type=Asset, required=True)
+
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        if (to_ts := data['to_timestamp']) <= (from_ts := data['from_timestamp']):
+            raise ValidationError(
+                message='from_timestamp must be smaller than to_timestamp',
+                field_name='from_timestamp',
+            )
+
+        if from_ts % (interval := data['interval']) != 0:
```

Comment:
why the multiples? Can't you just get to the closest one and return that? Both from and for to?